lib/rbvmomi/vim/Datastore.rb

```diff
@@ -63,6 +63,7 @@ def datacenter
   end
 
   def mkuripath path
-    "/folder/#{URI.escape path}?dcPath=#{URI.escape datacenter.name}&dsName=#{URI.escape name}"
+    datacenter_path_str = datacenter.path[1..-1].map{|elem| elem[1]}.join('/')
```

Comment:
Correct.  The documentation says we need the path.  The name alone only works as the path when the datacenter is in the root folder.  If the datacenter is in folderInner which is inside folderOuter the path would be 'folderOuter/folderInner/MyDatacenter'.  This was the simplest way I found to get that information.